.github/workflows/tests.yml

```diff
@@ -0,0 +1,78 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - master
+      - release
+  pull_request:
+    branches:
+      - '*'
+  schedule:
+    # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onschedule
+    # > Scheduled workflows run on the latest commit on the default or base branch
+    # i.e. this can only run on master
+    - cron:  '0 11 * * *'
+
+env:
+  zip_filepath: "batch_processing_results.zip"
+
+jobs:
+  batch_processing:
+    name: Test batch_processing.sh
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout SCT
+        uses: actions/checkout@v2
+
+      # install_sct edits ~/.bashrc, but those environment changes don't get passed to subsequent steps in GH Actions.
+      # So, we filter through the .bashrc and pass the values to $GITHUB_ENV and $GITHUB_PATH.
+      # Relevant documentation: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#environment-files
+      - name: Install SCT
+        run: |
+          PIP_PROGRESS_BAR=off ./install_sct -y
+          cat ~/.bashrc | grep "export SCT_DIR" | cut -d " " -f 2 >> $GITHUB_ENV
+          cat ~/.bashrc | grep "export MPLBACKEND" | cut -d " " -f 2 >> $GITHUB_ENV
+          cat ~/.bashrc | grep "export PATH" | grep -o "/.*" | cut -d ':' -f 1 >> $GITHUB_PATH
+
+      - name: Run batch_processing.sh
+        run: |
+          ./batch_processing.sh
+
+      # We use the REST API to fetch the most recently uploaded results from batch_processing.
+      # jq and awk are used to filter the JSON response by artifact ID, which is then used to download the right .zip.
+      # Relevant documentation: https://docs.github.com/en/rest/reference/actions#artifacts
+      - name: Fetch past results
+        run: |
+          export NEWEST_ARTIFACT_ID=$(curl -H "Accept: application/vnd.github.v3+json" \
+                                           https://api.github.com/repos/neuropoly/spinalcordtoolbox/actions/artifacts |
+                                      jq '.artifacts[].id' |
+                                      awk '{for(i=1;i<=NF;i++) if($i>maxval) maxval=$i;}; END { print maxval;}')
+          curl -u admin:${{ secrets.GITHUB_TOKEN }} \
```

Comment:
Is there a reason for passing username and password to this request? From the docs it looks like anyone with read access can request this endpoint (just like the `list` endpoint above).

Reply:
I haven't tested extensively, but locally when I try without `-u`, it gives me this response:

```bash
joshua@XPS-15-9560:~$ curl -H "Accept: application/vnd.github.v3+json" -L "https://api.github.com/repos/neuropoly/spinalcordtoolbox/actions/artifacts/36987543/zip"
{
  "message": "You must have the actions scope to download artifacts.",
  "documentation_url": "https://docs.github.com/rest/reference/actions#download-an-artifact"
}
```

So I think it's necessary to explicitly identify as a user in the request? I tested it locally using a personal access token and it worked OK.

It's hard to find documentation on this though -- the closest I've found for background context is the issue description in: https://github.com/actions/upload-artifact/issues/51#issue-558653519

> Likewise, when the Actions API was released, API requests worked without any authentication. One could simply query the API for an artifact download URL and redirect the guests to it. - All of that worked without GitHub account and could have worked from JavaScript or a small lightweight webservice which redirects the end-user to the latest artifact download through HTTP redirects (I wrote https://github.com/JayFoxRox/GitHub-artifact-URL for this purpose).
>
> However, within the last days these artifact download URLs were suddenly made private - they only work for registered GitHub users now. Everyone clicking the download button on our website (who isn't logged into GitHub) gets a confusing 404 error for direct artifact download URLs now.